File: Python/백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.py
import sys
input = sys.stdin.readline

N, M = map(int, input().split())
board = [input().rstrip() for _ in range(N)]

def count_repaint(sy, sx):
    cnt1 = 0  # W 시작 무늬로 맞출 경우
    cnt2 = 0  # B 시작 무늬로 맞출 경우
    for i in range(8):
        for j in range(8):
            current = board[sy + i][sx + j]
            if (i + j) % 2 == 0:
                # 이 칸은 시작 색과 같아야 함
                if current != 'W':
                    cnt1 += 1
                if current != 'B':
                    cnt2 += 1
            else:
                # 이 칸은 시작 색과 반대여야 함
                if current != 'B':
                    cnt1 += 1
                if current != 'W':
                    cnt2 += 1
    return min(cnt1, cnt2)

ans = float('inf')
for i in range(N - 7):
    for j in range(M - 7):
        ans = min(ans, count_repaint(i, j))

print(ans)

File: Java/백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
        char[][] board = new char[N][M];
        for (int i = 0; i < N; i++) {
            board[i] = br.readLine().toCharArray();
        }

        int ans = Integer.MAX_VALUE;
        for (int i = 0; i + 8 <= N; i++) {
            for (int j = 0; j + 8 <= M; j++) {
                ans = Math.min(ans, countRepaint(board, i, j));
            }
        }
        System.out.println(ans);
    }

    static int countRepaint(char[][] board, int sy, int sx) {
        int cnt1 = 0;  // W 시작
        int cnt2 = 0;  // B 시작
        for (int i = 0; i < 8; i++) {
            for (int j = 0; j < 8; j++) {
                char c = board[sy + i][sx + j];
                if ((i + j) % 2 == 0) {
                    if (c != 'W') cnt1++;
                    if (c != 'B') cnt2++;
                } else {
                    if (c != 'B') cnt1++;
                    if (c != 'W') cnt2++;
                }
            }
        }
        return Math.min(cnt1, cnt2);
    }
}